pkg/common/utils/utils_test.go

```diff
@@ -696,3 +699,356 @@ func TestNeedRecovery(t *testing.T) {
 		})
 	}
 }
+func TestGetTaskGroupFromPodSpec(t *testing.T) {
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pod-01",
+			UID:  "UID-01",
+			Annotations: map[string]string{
+				constants.AnnotationTaskGroupName: "test-task-group",
+			},
+		},
+	}
+
+	assert.Equal(t, GetTaskGroupFromPodSpec(pod), "test-task-group")
+
+	pod = &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pod-01",
+			UID:  "UID-01",
+		},
+	}
+
+	assert.Equal(t, GetTaskGroupFromPodSpec(pod), "")
+}
+
+// nolint: funlen
+func TestGetTaskGroupFromAnnotation(t *testing.T) {
+	// correct json
+	testGroup := `
+	[
+		{
+			"name": "test-group-1",
+			"minMember": 10,
+			"minResource": {
+				"cpu": 1,
+				"memory": "2Gi"
+			},
+			"nodeSelector": {
+				"test": "testnode",
+				"locate": "west"
+			},
+			"tolerations": [
+				{
+					"key": "key",
+					"operator": "Equal",
+					"value": "value",
+					"effect": "NoSchedule"
+				}
+			]
+		},
+		{
+			"name": "test-group-2",
+			"minMember": 5,
+			"minResource": {
+				"cpu": 2,
+				"memory": "4Gi"
+			}
+		}
+	]`
+	testGroup2 := `
+	[
+		{
+			"name": "test-group-3",
+			"minMember": 3,
+			"minResource": {
+				"cpu": 2,
+				"memory": "1Gi"
+			}
+		}
+	]`
+	// Error json
+	testGroupErr := `
+	[
+		{
+			"name": "test-group-err-1",
+			"minMember": "ERR",
+			"minResource": {
+				"cpu": "ERR",
+				"memory": "ERR"
+			},
+		}
+	]`
+	// without name
+	testGroupErr2 := `
+	[
+		{
+			"minMember": 3,
+			"minResource": {
+				"cpu": 2,
+				"memory": "1Gi"
+			}
+		}
+	]`
+	// without minMember
+	testGroupErr3 := `
+	[
+		{
+			"name": "test-group-err-2",
+			"minResource": {
+				"cpu": 2,
+				"memory": "1Gi"
+			}
+		}
+	]`
+	// withot minResource
+	testGroupErr4 := `
+	[
+		{
+			"name": "test-group-err-4",
+			"minMember": 3,
```

Comment:
This example is designed for nil minMenber.
So we should remove the comma, or this err will be classified to parsing error.
"minMember": 3,  -> "minMember": 3 

However, there is not an exception handling  for empty minResource in the GetTaskGroupsFromAnnotation function.

Reply:
Addressed in latest PR, thanks!